flink-cdc-e2e-tests/pom.xml

```diff
@@ -30,7 +30,7 @@ under the License.
     <packaging>jar</packaging>
 
     <properties>
-        <flink-1.13>1.13.3</flink-1.13>
+        <flink-1.14>1.14.2</flink-1.14>
```

Comment:
@wangxiaojing Thanks for the contribution, the user need the CDC Connector can run on both Flink 1.13 cluster and Flink 1.14 Cluster, thus I think what we should need to do is make the the CDC Connector can run on  Flink 1.14 Cluster, so we do not need to change the version, but only need to the incompatibility API that CDC Connector used, we can add e2e test with Flink 1.14 cluster to check this.